lib/src/local_storage.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:sembast/sembast.dart';
+
+import 'constants.dart';
+import 'cross_platform/path.dart' as path;
+import 'cross_platform/sembast.dart';
+
+class LocalStorage {
+  bool _isOpen = false;
+  late final Database _localDb;
+  late final StoreRef _store;
+
+  LocalStorage();
+
+  Future _openDb() async {
```

Comment:
Do we need to close the db after opening?

Reply:
That's a good point. Would we want to open & close it every time?

Reply:
> Opening a database means loading its content in memory. This is an expensive operation. In an application (flutter, web), the recommended way is to open the database on start and to keep it open during the lifetime of the application.

so we only want to close it when application exit and re-open on application start.

Reply:
I've done some looking, but I'm not sure of a way to do that (dart only - with no flutter). do you have any suggestions?

Reply:
should we use this package https://pub.dev/packages/hive instead of sembast? 

we can also support recoverSession automatically because `You can use Hive just like a map. It is not necessary to await Futures.`

Reply:
🤦 No idea how I missed that one......

It is perfect, I'll rewrite soon